packages/orchestration/src/exos/packet-tools.js

```diff
@@ -340,6 +340,7 @@ export const preparePacketTools = (zone, vowTools) => {
           const { tap } = this.facets;
           // XXX racy; fails if subscribeToTransfers is called while this promise is in flight
           // e.g. 'Target "agoric1fakeLCAAddress" already registered'
+          this.state.reg = 'RACING';
```

Comment:
Here's how I would solve the race: check for the cached registration, and if it's not present, atomically replace it with a vow.

```js
        subscribeToTransfers() {
          // Subscribe to the transfers for this account.
          const { lca, reg: cachedReg } = this.state;
          if (cachedReg) {
            return when(cachedReg);
          }
          // Atomically update the registration.
          const { tap } = this.facets;
          const reg = watch(E(lca).monitorTransfers(tap));
          this.state.reg = reg;
          return when(reg);
        },
```

Reply:
Oh, and the FIXME comment on line 330-331 is no longer needed:

```js
          // FIXME when it returns undefined this causes an error:
          // In "unsubscribeFromTransfers" method of (PacketToolsKit utils): result: undefined "[undefined]" - Must be a promise
```

It was solved 3 months ago by @0xpatrickdev changing `decrPendingPatterns`'s method guard to use `.returns(M.undefined())`.